File: src/IsbInterop/Scripts/IScriptFactory.vb
Imports IsbInterop.Base

Namespace Scripts
  ''' <summary>
  ''' Фабрика сценариев.
  ''' </summary>
  Public Interface IScriptFactory
    Inherits IFactory(Of IScript, IObjectInfo)

    ''' <summary>
    ''' Получить сценарий по имени.
    ''' </summary>
    ''' <param name="scriptName">Имя сценария.</param>
    ''' <returns>Сценарий.</returns>
    Function GetObjectByName(scriptName As String) As IScript
  End Interface
End NameSpace

File: src/IsbInterop.CSharp/Scripts/IScriptFactory.cs
using IsbInterop.Base;

namespace IsbInterop.Scripts
{
  /// <summary>
  /// Фабрика сценариев.
  /// </summary>
  public interface IScriptFactory : IFactory<IScript, IObjectInfo>
  {
    /// <summary>
    /// Получает сценарий по имени.
    /// </summary>
    /// <param name="scriptName">Имя сценария.</param>
    /// <returns>Сценарий.</returns>
    IScript GetObjectByName(string scriptName);
  }
}
